src/core/aio/disk_engine.cpp

```diff
@@ -138,12 +142,9 @@ disk_engine::disk_engine()
 {
     _node = service_engine::instance().get_all_nodes().begin()->second.get();
 
-    // use native_linux_aio_provider in default
-    if (!strcmp(FLAGS_aio_factory_name, "dsn::tools::sim_aio_provider")) {
-        _provider.reset(new aio::sim_aio_provider(this, nullptr));
-    } else {
-        _provider.reset(new native_linux_aio_provider(this, nullptr));
-    }
+    aio_provider *provider = utils::factory_store<aio_provider>::create(
+        FLAGS_aio_factory_name, dsn::PROVIDER_TYPE_MAIN, this, nullptr);
```

Comment:
If FLAGS_aio_factory_name is wrong, provider will be nullptr.Is any protection for null _provider?

Reply:
I can add a assert here.